armi/reactor/blocks.py

```diff
@@ -1757,38 +1767,43 @@ def rotatePins(self, rotNum, justCompute=False):
             raise ValueError(
                 "Cannot rotate {0} to rotNum {1}. Must be 0-5. ".format(self, rotNum)
             )
-        # pin numbers start at 1.
-        numPins = self.getNumComponents(Flags.CLAD)  # number of pins in this assembly
+
+        # Pin numbers start at 1. Number of pins in the block is assumed to be based on
+        # cladding count.
+        numPins = self.getNumComponents(Flags.CLAD)
         rotateIndexLookup = dict(zip(range(1, numPins + 1), range(1, numPins + 1)))
 
-        # look up the current orientation and add this to it. The math below just rotates from the
-        # reference point so we need a total rotation.
+        # Look up the current orientation and add this to it. The math below just rotates
+        # from the reference point so we need a total rotation.
         rotNum = int((self.getRotationNum() + rotNum) % 6)
 
         # non-trivial rotation requested
         # start at 2 because pin 1 never changes (it's in the center!)
         for pinNum in range(2, numPins + 1):
             if rotNum == 0:
-                # rotation to reference orientation. Pin locations are pin IDs.
+                # Rotation to reference orientation. Pin locations are pin IDs.
                 pass
             else:
-                # Determine the pin ring (courtesy of Robert Petroski from subchan.py). Rotation does not change the pin ring!
+                # Determine the pin ring. Rotation does not change the pin ring!
                 ring = int(
                     math.ceil((3.0 + math.sqrt(9.0 - 12.0 * (1.0 - pinNum))) / 6.0)
                 )
+
                 # Rotate the pin position (within the ring, which does not change)
                 tot_pins = 1 + 3 * ring * (ring - 1)
                 newPinLocation = pinNum + (ring - 1) * rotNum
                 if newPinLocation > tot_pins:
                     newPinLocation -= (ring - 1) * 6
+
                 # Assign "before" and "after" pin indices to the index lookup
                 rotateIndexLookup[pinNum] = newPinLocation
 
-            if not justCompute:
-                self.p["pinLocation", rotateIndexLookup[pinNum]] = pinNum
-
+        # Because the above math creates indices based on the absolute rotation number,
+        # the old values of pinLocation (if they've been set in the past) can be overwritten
+        # with new numbers
         if not justCompute:
             self.setRotationNum(rotNum)
+            self.p["pinLocation"] = [*rotateIndexLookup.values()]
```

Comment:
`roateIndexLookup.values()` will probably be sorted in ascending order of the keys... but should we code it explicitly to be sorted this way? For example:

`self.p["pinLocation"] = [v for k, v in sorted(rotateIndexLookup.items())]`

Reply:
Or
`self.p["pinLocation"] = [ rotateIndexLookup[pinNum] for pinNum in range(1, numPins + 1) ]`

Reply:
I think your way is more robust. I believe `values()` returns based on the order of how keys are added to the dictionary since Python maintains dictionary order based on how the keys are added to the dictionary. Since `rotateIndexLookup` is declared with the desired order, then `keys()` and `values()` will maintain that order. But if someone were to edit that declaration, then `values()` may no longer yield the desired result.

I will make this change.